scripts/postcss/README.md

```diff
@@ -0,0 +1,99 @@
+# Terra Postcss Theme plugin
+
+The purpose of this plugin is to create a default theme from a scoped theme and to remove any known themes that are not desired.
+
+The plugin will read from the same terra-theme.config.js that terra-aggregate-themes script reads from. Eventually terra-aggregate-themes will be deprecated in favor of this strategy.
+
+## Configuration
+
+### terra-theme.config.js
+
+Below is an example of terra-theme.config
+
+```js
+const themeConfig = {
+  exclude: [], // Files to exclude. Accepts glob patterns.
+  theme: 'terra-dark-theme', // The default theme.
+  scoped: ['terra-light-theme', 'terra-lowlight-theme'], // An array of scoped themes.
+};
+
+module.exports = themeConfig;
+```
+
+## Usage
+
+To use this plugin you must have strong conventions around your theme name, include the theme class in your css and add the postcss plugin.
+
+### CSS
+
+This plugin makes the assumption that you are declaring theme variables under a global css class i.e. ```.orion-fusion-theme``` and these theme files are included in such a way that they are processed by webpack.
+
+Consider the following component. The react component pulls in and applies the css styles to div.
+
+Component.jsx
+
+```jsx
+import React from 'react';
+import { ThemeContext } from 'terra-application/lib/theme';
+
+import styles from './component.module.scss';
+
+const cx = classNames.bind(styles);
+
+const Component = ({children}) => {
+  const theme = React.useContext(ThemeContext);
+  return (
+    <div className={cx('div', theme.className)}>
```

Comment:
For this new theme strategy, does this mean all components will need a wrapping div to correctly flip themes? There are times we use React.Fragement to group component renders. Will we need to apply the global theme name to each element in those cases?

Reply:
If we're applying styles we have some DOM element we're attaching those styles to. I would expect the uppermost DOM element in the component would have the styles applied to it. Or you could apply appropriate theme variables to each DOM element provided in some sort of react fragment. If your component doesn't have any DOM elements it probably doesn't need themeing :)